staticDHCPd/extensions/official/httpdb.py

```diff
@@ -263,12 +266,34 @@ def lookupMAC(self, packet_or_mac, packet_type=None, mac=None, ip=None,
 
 class HTTPCachingDatabase(CachingDatabase, _HTTPLogic):
     def __init__(self):
+        from staticdhcpdlib import config
         if hasattr(config, 'X_HTTPDB_CONCURRENCY_LIMIT'):
             CachingDatabase.__init__(self, concurrency_limit=config.X_HTTPDB_CONCURRENCY_LIMIT)
         else:
             CachingDatabase.__init__(self)
         _HTTPLogic.__init__(self)
 
+    def lookupMAC(self, packet_or_mac, packet_type=None, mac=None, ip=None,
+                  giaddr=None, pxe_options=None):
+        cache_mac = packet_or_mac if type(packet_or_mac) == MAC else mac
+
+        if self._cache and cache_mac:
+            try:
+                definition = self._cache.lookupMAC(cache_mac)
+            except Exception, e:
```

Comment:
Also, we could probably transform this file to use the newer 'as exc' syntax and not use the traceback module